packages/tools-packages/src/types.ts

```diff
@@ -0,0 +1,36 @@
+import type { PackageManifest } from "@rnx-kit/tools-node";
+
+export type PackageInfo = {
+  /** name of the package */
+  name: string;
+
+  /** full path to the package */
+  root: string;
+
+  /** Access the loaded package.json for the package */
+  manifest: PackageManifest;
+
+  /** Is this a workspace package */
+  workspace: boolean;
+
+  /**
+   * data storage by symbol value is allowed to have package specific values stored here. This allows
+   * other packages to leverage any caching happening for PackageInfo entries to store additional
+   * information and ensure it is only loaded once.
+   */
+  [key: symbol]: unknown;
```

Comment:
Is it possible to replace `unknown` with `T` to avoid casting in `createPackageValueLoader` and `createPackageValueAccessors`?
```suggestion
  [key: symbol]: T;
```

Reply:
It isn't because the T isn't the same for all entries. That's part of the reason the accessors have the symbol contained, the type-casting is safe because no one else can get at that value.

The other option would be to have this be a type union but you can't anticipate all things people might want to use. If you did try the typed variant where you made something like PackageInfo<T = unknown> then the accessors would have to do the typecasting on the way in, which is messier to me.